cds/modules/maintenance/cli.py

```diff
@@ -38,13 +37,50 @@
     create_subformat,
 )
 from cds.modules.records.api import CDSVideosFilesIterator
+from cds.modules.deposit.api import Video
 from cds.modules.records.resolver import record_resolver
+from cds.modules.deposit.tasks import datacite_register
+from invenio_db import db
+from cds.modules.records.permissions import is_public
+from invenio_indexer.api import RecordIndexer
+
 
 
 def abort_if_false(ctx, param, value):
     if not value:
         ctx.abort()
 
+@click.command()
+@click.option("--recid", "recid", help="ID of the video record", default=None, required=True)
+@with_appcontext
+def create_doi(recid):
+    try:
+        # Get the video object with recid
+        _, record = record_resolver.resolve(recid)
+        depid = record.depid
+        video_deposit = Video.get_record(depid.object_uuid)
+    except Exception as exc:
```

Comment:
Can we maybe keep only one `except`? You could have multiple exceptions caught and I would restructure a bit the order of commands like below:
```python
_, record = record_resolver.resolve(recid)
depid = record.depid
video_deposit = Video.get_record(depid.object_uuid)
try:
   # Mint the doi and publish
    edit_record = video_deposit.edit().mint_doi().publish().commit()
    # Save changes
    db.session.commit()
   
   # Index the record again
    _, record_video = edit_record.fetch_published()
    RecordIndexer().index(record_video)
    click.echo(f"DOI minted and indexed successfully for record '{recid}'")
    
    datacite_register(recid, str(record.id))

except DataCiteError as dexc:
    raise ClickException(f"Failed to register DOI on datacite for the video record '{recid}': {str(e)}")
except Exception as e:
    db.session.rollback()
    raise ClickException(f"Failed to mint DOI for the video record '{recid}': {str(e)}")
```
- I would put the datacite registration after the video metadata publish and index because if it fails we can just retry that part
- You can use the datacite specific exception https://github.com/inveniosoftware/datacite/blob/master/datacite/errors.py#L24 to catch only the datacite related errors